### PR TITLE
Added Protractor UI and Functionality Tests and also Added Backend Test Script and API Endpoint tests For Bassa

### PR DESCRIPTION
**Pull request by iammosespaulr**

### CODE REVIEW

**Comment by iammosespaulr** [comment]
@agentmilindu please review ...
screenshot from 2018-01-15 00-20-15

File: components/core/Test.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from UserManager import * 
+import unittest
+username = "admin"
+password = "admin"
+usernamer = "rand"
+passwordr = "pass"
+class Test(unittest.TestCase):
```
Comment (malithsen) [inline components/core/Test.py]: Let's use function docstrings for each of the test cases

File: components/core/Test.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
+from UserManager import *
```
Comment (malithsen) [inline components/core/Test.py]: We already have a components/core/test directory, let's contain all the test case there.

Have meaningful test file names. eg: test_login.py

**Comment by iammosespaulr** [comment]
@malithsen @agentmilindu please review

File: components/core/test_login.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+def username():
+ return "admin"
+def password():
+ return "admin"
+def usernamer():
+ return "rand"
+def passwordr():
+ return "pass"
+class Test(unittest.TestCase):
+ def test_Incorrect_Login(self):
```
Comment (JaDogg) [inline components/core/test_login.py]: function names should be snake case

**Comment by iammosespaulr** [comment]
@JaDogg The reason why i used functions for username is because @malithsen asked me to do so .. "
Let's use function docstrings in each of the test cases"  and Aren't the function names in snake case... ??

File: components/core/test_login.py (hunk 3 above)
Comment (iammosespaulr) [reply components/core/test_login.py]: they are in snake case aren't they ??

**Comment by JaDogg** [comment]
@iammosespaulr Do not mix capital and simple in test case names. 👎 Username function is OK. 👍 Nicely done. 😄

**Comment by iammosespaulr** [comment]
Ok @JaDogg Will do the Capital Change :smile:

**Comment by iammosespaulr** [comment]
@JaDogg made the capital change ..

**Comment by iammosespaulr** [comment]
I've Also Added Protractor UI tests ...
screenshot from 2018-01-21 12-45-48

**Comment by iammosespaulr** [comment]
@agentmilindu @malithsen @JaDogg @rehrumesh  please review

**Comment by iammosespaulr** [comment]
Fixes Issue #418

File: components/core/Test.py (hunk 1 above)
Comment (iammosespaulr) [reply components/core/Test.py]: Done ..

**Comment by agentmilindu** [comment]
@malithsen If you can review the new changes again and approve, we can merge this PR.

File: components/core/Test.py (hunk 2 above)
Comment (iammosespaulr) [reply components/core/Test.py]: done

**Comment by iammosespaulr** [comment]
I have also added API endpoint tests #414 i will add that file as well and then we can merge the PR .. WDYS ?

**Comment by iammosespaulr** [comment]
I will also try to fix existing ESlint errors in the bassa repository

**Comment by iammosespaulr** [comment]
@malithsen please review nd merge ..

**Comment by malithsen** [comment]
Please look into the failing Codacy and Travis builds

**Comment by iammosespaulr** [comment]
@malithsen im not sure whats happening with the Travis-ci ... could you help me with that .. and  i am working on fixing the codacy issues

**Comment by iammosespaulr** [comment]
@agentmilindu @rehrumesh help me fi the travis ci build ... im not sure what the problem is

**Comment by iammosespaulr** [comment]
Yayy @malithsen i found out how to fix the travis ci builld ... will do the codacy changes as well

**Comment by malithsen** [comment]
@iammosespaulr Good work :+1: 
It would be nice if we can integrate these new tests with Travis

**Comment by iammosespaulr** [comment]
@malithsen yeah could you help me with that ...

**Pull request merged**